python/ray/data/_internal/plan.py

```diff
@@ -601,12 +612,12 @@ def clear_block_refs(self) -> None:
 
     def stats(self) -> DatasetStats:
         """Return stats for this plan, forcing execution if needed."""
-        self.execute()
+        if not self._snapshot_stats:
```

Comment:
Should we just return self._snapshot_stats instead of forcing execution (i.e. None if execution hasn't started)?
The `ds.stats()` is an inquiry (like const method) and we could return whatever already there?